lib/rucio/daemons/rsedecommissioner/throttling.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes used for throttling RSE decommissioning."""
+
+import logging
+from typing import TYPE_CHECKING
+
+from rucio.db.sqla.constants import RuleState
+
+if TYPE_CHECKING:
+    from typing import Any
+    from collections.abc import Callable
+    from rucio.db.sqla import models
```

Comment:
I'm actually confused about the standard practice for type hinting in rucio. At some point I got the impression that stringized annotations are favored, but reading the [style guide](https://rucio.cern.ch/documentation/developer/type_annotation_guide) (which I should have read much sooner; I'm sorry) it says otherwise. Should I revert to annotating as e.g. `dict[str, Any]` instead of `'dict[str, Any]'`? Or is the point that `dict[]` must be stringized becaus it's not compatible with python < 3.9?

Reply:
OK, thank you for the clarification. Doing as such now.